### PR TITLE
Use SVG external content for social-logos.

### PR DESCRIPTION
**Pull request by sgomes**
Reimplement social-logos using an SVG external content approach, instead of SVG transpiled into React components. This helps keep images as images (which is good for performance), while maintaining styleability. Old browsers are supported through the already-included `svg4everyone` polyfill.

See #32364 for a similar change, which added Material Icons with the same approach.

This PR moves ~15KB from JS to SVG in several sections and async chunks.

#### Changes proposed in this Pull Request

* Bump `social-logos` package version to `2.1.0`, which includes the SVG spritemap.
* Implement `components/social-logo` using SVG external content approach.
* Rewrite all uses of `SocialLogo` to use `components/social-logo` instead of the `social-logos` package.
* Replace usage of example in `social-logos` package with a new example in `components/social-logo/docs`.

#### Testing instructions

* Test as many places where social logos are used as possible. Ensure that logos look identical to before this change.

### CODE REVIEW

**Comment by ockham** [comment]
Devdocs still pull in example content from the package: https://github.com/Automattic/wp-calypso/blob/48b121e7ce115566888e350b9deae4a47e6d1a0f/client/devdocs/design/index.jsx#L104

(using this alias: https://github.com/Automattic/wp-calypso/blob/48b121e7ce115566888e350b9deae4a47e6d1a0f/webpack.config.js#L252)

**Comment by sgomes** [comment]
> Devdocs still pull in example content from the package:
> 
> https://github.com/Automattic/wp-calypso/blob/48b121e7ce115566888e350b9deae4a47e6d1a0f/client/devdocs/design/index.jsx#L104
> 
> (using this alias:
> 
> https://github.com/Automattic/wp-calypso/blob/48b121e7ce115566888e350b9deae4a47e6d1a0f/webpack.config.js#L252
> 
> )

Yes, I intentionally left it that way, since I assume not a lot of regular users use devdocs, and since the example actually lives outside this repo, inside `social-logos`.

We could of course reimplement the example in Calypso using the new approach. Do you think that would be best?

**Comment by sgomes** [comment]
Currently investigating test failure. It seems the transformation we use in tests to enable compatibility with `file-loader` isn't working here, for some reason.

**Comment by sgomes** [comment]
@jsnajdr 

> Currently investigating test failure. It seems the transformation we use in tests to enable compatibility with `file-loader` isn't working here, for some reason.

The latest commit should fix this. `social-logos` was being excluded from the jest transform that enables `file-loader` compatibility.

**Comment by ockham** [comment]
> > Devdocs still pull in example content from the package:
> > https://github.com/Automattic/wp-calypso/blob/48b121e7ce115566888e350b9deae4a47e6d1a0f/client/devdocs/design/index.jsx#L104
> > 
> > (using this alias:
> > https://github.com/Automattic/wp-calypso/blob/48b121e7ce115566888e350b9deae4a47e6d1a0f/webpack.config.js#L252
> > 
> > )
> 
> @ockham: Yes, I intentionally left it that way, since I assume not a lot of regular users use devdocs, and since the example actually lives outside this repo, inside `social-logos`.
> 
> We could of course reimplement the example in Calypso using the new approach. Do you think that would be best?

Maybe -- but then again, it's really not that critical :+1: 

I was mostly thinking about a future `social-logos` package that drops the JS wrapper altogether and only ships the SVG. Not sure if it would/should then still ship a pre-built example, or if that should be Calypso's/DevDocs' concern :thinking:

Arguably, it should be consistent with Gridicons, which currently do have their own (Calypso-independent) [website](http://automattic.github.io/gridicons/) :thinking:

**Comment by sgomes** [comment]
> Arguably, it should be consistent with Gridicons, which currently do have their own (Calypso-independent) [website](http://automattic.github.io/gridicons/) 🤔

I suppose at that point you could consider it two separate examples: usage of the default SVG-in-JS icons shipped with the packages, and usage of the Calypso wrapper for the icons. And if we were ever to ship `social-logos` as just the SVG, we could reevaluate.

I'll implement an example in Calypso using the new approach, then 🙂

**Comment by sgomes** [comment]
@ockham Removed the dependency on the `social-logos` package example and added a new example in Calypso.

File: client/devdocs/design/component-examples.js
Comment (sgomes) [inline client/devdocs/design/component-examples.js]: Note: these changes were made to appease ESLint, which doesn't handle `eslint-disable` correctly for parsing errors, such as the one the previous syntax caused.

File: client/devdocs/design/playground-scope.js
Comment (sgomes) [inline client/devdocs/design/playground-scope.js]: Note: these changes were made to appease ESLint, which doesn't handle `eslint-disable` correctly for parsing errors, such as the one the previous syntax caused.

File: client/my-sites/marketing/buttons/preview-button.jsx
Comment (sgomes) [reply client/my-sites/marketing/buttons/preview-button.jsx]: Good catch! Fixed.

**Comment by jsnajdr** [comment]
#32761 should improve the TypeScript linting situation. `export` extensions in JS files should stop breaking things, and there is only one `no-unused-vars` rule.

**Comment by sgomes** [comment]
> #32761 should improve the TypeScript linting situation. `export` extensions in JS files should stop breaking things, and there is only one `no-unused-vars` rule.

I'll put this PR on hold until #32761 gets merged. Once it does, I'll remove all the ESLint-related changes. Feel free to either review now with that in mind, or wait until then to review!

**Comment by sgomes** [comment]
This PR has now been rebased on top of #32761 and is ready for a final review!

File: client/devdocs/design/component-examples.js
Comment (sgomes) [reply client/devdocs/design/component-examples.js]: Outdated.

File: client/devdocs/design/playground-scope.js
Comment (sgomes) [reply client/devdocs/design/playground-scope.js]: Outdated.

**Review by blowery — APPROVED** [review]
Looks good to me! Nice work.

**Comment by sgomes** [comment]
Thanks for the review, @blowery! 👍 I'll wait for @jsnajdr's ok too before merging.

**Pull request merged**
